src/qt/dc.cpp

```diff
@@ -115,14 +115,50 @@ bool wxQtDCImpl::CanGetTextExtent() const
 
 void wxQtDCImpl::DoGetSize(int *width, int *height) const
 {
-    if (width)  *width  = m_qtPainter->device()->width();
-    if (height) *height = m_qtPainter->device()->height();
+    QPaintDevice *pDevice = m_qtPainter->device();
+
+    int deviceWidth;
+    int deviceHeight;
+
+    if ( pDevice )
+    {
+        deviceWidth = pDevice->width();
+        deviceHeight = pDevice->height();
+    }
+    else
+    {
+        deviceWidth = 0;
```

Comment:
I'm not sure it's a good idea to silently return (0,0) for an invalid DC. Is it really normal for a DC to be invalid in wxQt? This isn't the case in the other ports, AFAIK.

In which circumstances exactly does the crash due to the invalid `QPaintDevice` happen?

Reply:
In wxQT if you use a memory DC without first calling SelectObject,  or by calling SelectObject with a "null" bitmap.  Then the QPainter will have a NULL QPaintDevice.

Reply:
Could it be better to do this check in `wxMemoryDCImpl` only? For it we clearly do need it, as in the situation you describe `wxMemoryDC::GetSize()` returns (0,0) under the other platforms too, but for other kinds of DC I'm not so sure...

Oh well, I guess we can always do it later.